### PR TITLE
[FIX]: Cluster dashboard not rendered

### LINKED ISSUE #811: Cluster dashboard not rendered
**Describe the bug**
For some reason, cluster dashboard is not rendered when opening the cluster. The issue occurred after connection to the cluster was lost due network interruption and tried to reconnect.

image

image

Edit: seems that there are more than one identical iframe element in the DOM:
image

### PR DESCRIPTION
**Pull request by ixrock**
Current solution: to keep iframes in DOM (reseting `iframe.src` should free up used memory afaik)

image

close #811

### CODE REVIEW

File: src/main/cluster.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export class Cluster implements ClusterModel {
     if (!this.eventDisposers.length) {
       this.bindEvents();
     }
-    if (this.disconnected || !this.accessible) {
+    if (this.disconnected) {
```
Comment (nevalla) [inline src/main/cluster.ts]: Why `!this.accessible` is removed? Is it causing some issues?

**Comment by aleksfront** [comment]
I don't know how it's possible but `const rootElem = document.getElementById("app")` inside `src/renderer/bootstrap.tsx` gets wrong element. Not the one inside current iframe, but first found from the root DOM. Maybe it's possible to fine-tune dom selector somehow to solve the issue.

**Comment by ixrock** [comment]
@aleksfront problem of the issue not in DOM at all :)
`webFrame.routingId` returns cached id for previously created iframe with same url or something..
In that case cluster-view didn't receive proper state after `refresh()` and thus not rendered at all.

File: src/main/cluster.ts (hunk 1 above)
Comment (ixrock) [reply src/main/cluster.ts]: Yes, it's trying to re-connect on initial cluster activation. Added extra check, now seems to work fine.

**Review by Nokel81 — APPROVED** [review]
Looks good

File: src/renderer/components/cluster-manager/lens-views.ts [hunk 2]
```diff
@@ -21,29 +21,38 @@ export async function initView(clusterId: ClusterId) {
...
+  iframe.removeAttribute("name")
 }
 
 export function refreshViews() {
   const cluster = getMatchedCluster();
   lensViews.forEach(({ clusterId, view, isLoaded }) => {
-    const isVisible = cluster && cluster.available && cluster.id === clusterId;
-    view.style.display = isLoaded && isVisible ? "flex" : "none"
+    const isCurrent = clusterId === cluster?.id;
+    const isReady = cluster?.available && cluster?.ready;
```
Comment (jim-docker) [inline src/renderer/components/cluster-manager/lens-views.ts:54]: this line could come outside of the `forEach()` call

**Review by nevalla — APPROVED** [review]
LGTM

**Pull request merged**